Test `*risk_category` in `sector_profile()` at product level

maurolepore opened the pull request:
(Note "sector profile" is the new name for "pstr".)

In an earlier version of the package I see a test for the internal function `prepare_inter_pstr_product()`:

```r
test_that("risk_category colummn should not have more than one NA for no result companies", {
  out <- prepare_inter_pstr_product(pstr_product, ep_companies, ecoinvent_activities, matches_mapper) |>
    filter(is.na(risk_category)) |>
    group_by(companies_id) |>
    summarise(count = n())
  
  expect_equal(unique(out$count), 1L)
})
```

This PR restores that test but via the public interface: `prepare_pstr_product()`. My inspection suggests that `prepare_pstr_product()` does nothing with `NA`s so they should be exactly as they were in the intermediate state created by the internal `prepare_inter_pstr_product()`.

Note I use a small version of matches mapper to reduce computation time. I see the large and small versions of that file both have only 1 NA in the output of `prepare_*pstr_product()`.

@kalashsinghal please revew. As the author of the original test you know best why that test is important and to understand if it makes sense to test this via `prepare_pstr_product()` rather than via the internal `prepare_inter_pstr_product()`.

kalashsinghal reviewed (changes requested):
@maurolepore it's alright to test this issue via `prepare_pstr_product`. 

`prepare_inter_pstr_product` function is used in both `prepare_pstr_product` and `prepare_pstr_company` functions. Hence, to avoid duplicating the test in test files of both of these functions I created a separate test file for `prepare_inter_pstr_product`. 

Please make sure that this test is also included for `prepare_pstr_company` function. Thanks :)

maurolepore commented:
Thanks @kalashsinghal, 

* I added an equivalent test for `prepare_pstr_company()`.
* I changed the expectation to better reflect the intent expressed in the title ("no more than 1" is equivalent to  "less than or equal to 1" -- which can be implemented with `expect_lte()`).
* I changed the input data to clarify the focus is on "empty" tiltIndicator results.

kalashsinghal reviewed (approved):
@maurolepore LGTM :)
